### DISCUSSION TITLE
Error when solving optimal control

### DISCUSSION THREAD

**Discussion by PrakitrSrisuma** [casadi/casadi#3282] [Q&A]
Hello,

I received the following errors after trying to solve an optimal control problem with Direct Shooting in CasADi.
I tried many things, but it still does not work. I am not sure if someone can help. Thanks a lot!

WARNING("IpoptUserClass::eval_h failed:.../casadi/core/oracle_function.cpp:285: Error in solver:nlp_hess_l:Error in Function::operator() for 'nlp_hess_l' [MXFunction] at .../casadi/core/function.cpp:1482:
Error in Function::operator() for 'fwd2_adj1_F' [MXFunction] at .../casadi/core/function.cpp:1482:
Error in Function::operator() for 'fwd2_asens1_F' [MXFunction] at .../casadi/core/function.cpp:1482:
Error in Function::operator() for 'fsens2_asens1_F' [CvodesInterface] at .../casadi/core/function.cpp:1482:
.../casadi/interfaces/sundials/cvodes_interface.cpp:392: CVodeF returned "CV_ERR_FAILURE". Consult CVODES documentation.") [.../casadi/interfaces/ipopt/ipopt_nlp.cpp:200]

Prakitr

**Comment by brunoml5** [comment]
Hi! Looks like the CVODES integrator is not converging. Try analysing your ODE to see if there are any tricky expressions such as sqrt or division by zero.
